### PR TITLE
replace 'vars' with 'spec_vars' (close #436)

### LINKED ISSUE #436: Re-definition found for builtin function `vars()`
`string_evaluator.py`, `string_evaluator.py`, `spec_evaluator.py` uses `vars` as a function argument. 

Since `vars()` is a builtin keyword. It's generally an anti-pattern to not redefine builtin function. We would need to refactor and update the definition. 

Static Analysis Issue [Link](https://deepsource.io/gh/scanapi/scanapi/issue/PYL-W0622/occurrences?page=1) to check the particular code line.

### PR DESCRIPTION
**Pull request by jhermann**
Implements #436 

Since vars() is a builtin keyword. It's generally an anti-pattern to redefine builtin functions.

### CODE REVIEW

**Comment by Pradhvan** [comment]
@camilamaia I think we can ignore the DeepSource failing for this PR since it's complaining about missing module/function docstrings and class docstrings.

**Review by Pradhvan — APPROVED** [review]
LGTM ⭐ Thank you for this effort.

File: tests/unit/tree/endpoint_node/test_validate.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def test_should_call_validate_keys(self, mock_validate_keys):
                 "path",
                 "requests",
                 "delay",
-                "vars",
+                tree_keys.VARS_KEY,
```
Comment (camilamaia) [inline tests/unit/tree/endpoint_node/test_validate.py:35]: good catch! 👍

**Review by camilamaia — APPROVED** [review]
⭐️

**Pull request merged**